src/wallet/wallet.cpp

```diff
@@ -3007,6 +3015,7 @@ bool CWallet::AttachChain(const std::shared_ptr<CWallet>& walletInstance, interf
                 return false;
             }
         }
+        walletInstance->m_attaching_chain = false;
```

Comment:
If you never started a rescan, than you would leave `m_attaching_chain` in true state forever

Reply:
Oh yes, thanks. Moved the spot where we set `m_attaching_chain` into the conditional.